lib/active_record/with_eager_group.rb

```diff
@@ -9,7 +9,7 @@ def exec_queries
     end
 
     def eager_group(*args)
-      check_if_method_has_arguments!(__callee__, args)
+      check_argument_not_blank!(args)
```

Comment:
please leave a comment to explain why you use @check_argument_not_blank!@ rather than @check_if_method_has_arguments!@

Reply:
`check_if_method_has_arguments!` would flatten all the arguments!  which would cause `[:eager_group_definition, scope_arg1, scope_arg2]` not able to preload together with other eager_group_definitions.


An example, I'd like to preload multiple different counts in one query.
```
class User
  has_many :students
  has_many :posts
 
  define_eager_group :posts_count, :posts, :count, :*
  define_eager_group :released_posts_count, :posts, :count, :*, -> { released }
  define_eager_group :reviewed_posts_count, :posts, :count, :*, -> (reviewer) { reviewed_by(reviewer) } 
end

class Student
  has_many :posts

  define_eager_group :posts_count, :posts, :count, :*
end

class Post
  belongs_to :user
  belongs_to :student

  scope :released, -> { where(released: true) }
  scope :reviewed_by, -> (user) { where(:reviewed_by_id => user.id) } 
end

User.eager_group(
  :posts_count, 
  :released_posts_count, 
  [:reviewed_posts_count, User.find(1)], 
  students: [:posts_count]
)

# with check_argument_not_blank!
args = [:posts_count, :released_posts_count, [:reviewed_posts_count, User.find(1)], students: [:posts_count]]

#with check_if_method_has_arguments!
args = [:posts_count, :released_posts_count, :reviewed_posts_count, User.find(1), students: [:posts_count]]

```
We can see the difference is `User.find(1)` become same level as other eager_group definitions, it lost its data structure, and would be treated as an eager_group definition.
Later we would receive an `Argument Error: wrong number of arguments (x for x)`

I added an unit test in this PR, you can test it without my code changes and find out the issue.